### PR TITLE
lightning: do resolve conflict job when other lightning has local dupes

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40923

Problem Summary:

### What is changed and how it works?
do resolve conflict job when other lightning has local dupes.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below
Manully test failed ti-case.

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix Lightning may skip resolving conflicts when other lightning meets local dupes while the last one doesn't.
```

### CODE REVIEW

File: br/tests/lightning_duplicate_resolution_incremental/run.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+LOG_FILE1="$TEST_DIR/lightning-duplicate-resolution1.log"
+LOG_FILE2="$TEST_DIR/lightning-duplicate-resolution2.log"
+
+# let lightning run a bit slow to avoid some table in the first lightning finish too fast.
+export GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/lightning/restore/SlowDownCheckDupe=sleep(5000)"
+run_lightning --backend local --sorted-kv-dir "$TEST_DIR/lightning_duplicate_resolution_incremental.sorted1" \
+  --enable-checkpoint=1 --log-file "$LOG_FILE1" --config "tests/$TEST_NAME/config1.toml" &
+run_lightning --backend local --sorted-kv-dir "$TEST_DIR/lightning_duplicate_resolution_incremental.sorted2" \
+  --enable-checkpoint=1 --log-file "$LOG_FILE2" --config "tests/$TEST_NAME/config2.toml" &
```
[the remark this answers is not in the document]
Comment (gozssky) [reply br/tests/lightning_duplicate_resolution_incremental/run.sh:29]: From my understanding, only the first lightning has duplicates. We need to let the second lightning detect the duplicate from the first lightning.

[the remark this answers is not in the document]
Comment (lichunzhu) [reply br/tests/lightning_duplicate_resolution_incremental/run.sh:29]: Nice catch. I will take a try

[the remark this answers is not in the document]
Comment (lichunzhu) [reply br/tests/lightning_duplicate_resolution_incremental/run.sh:46]: addressed in https://github.com/pingcap/tidb/pull/41157/commits/a4ebdeef63a4dbdb86e2dc84ce22f0a601b7f79a

**Review by dsdashun — APPROVED** [review]

File: br/tests/lightning_duplicate_resolution_incremental/run.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (lichunzhu) [reply br/tests/lightning_duplicate_resolution_incremental/run.sh:46]: @gozssky PTAL

**Review by gozssky — APPROVED** [review]

**Pull request merged**